app/src/main/java/com/eventyay/organizer/core/track/create/CreateTrackFragment.java

```diff
@@ -72,19 +71,25 @@ public void onStart() {
     }
 
     private void setColorPicker() {
-        if (colorPickerDialog == null)
-            colorPickerDialog = new ColorPicker(getActivity(), createTrackViewModel.getRed(), createTrackViewModel.getGreen(), createTrackViewModel.getBlue());
-
         binding.form.colorPicker.setBackgroundColor(createTrackViewModel.getColorRGB());
 
         binding.form.colorPicker.setOnClickListener(view -> {
-            colorPickerDialog.show();
-        });
-
-        colorPickerDialog.setCallback(color -> {
-            binding.form.trackColor.setText(String.format("#%06X", 0xFFFFFF & color));
-            binding.form.colorPicker.setBackgroundColor(color);
-            colorPickerDialog.dismiss();
+            new ColorPickerPopup.Builder(getContext())
```

Comment:
How about adding this colorPicker listener as a method in a related class and then reusing it in both CreateTrackFragment and UpdateTrackFragment?

Reply:
We can shift it to Utils but I don't think there is a need for that since even earlier it was done like this only.